internal/handlers/base.go

```diff
@@ -56,7 +66,101 @@ func GetURL(storage storage.Storage) http.HandlerFunc {
 			http.Error(res, "Not Found", http.StatusNotFound)
 			return
 		}
+		result, _ := storage.IsDeleted(shortURL)
+		if result {
+			res.WriteHeader(http.StatusGone)
+			return
+		}
 		res.Header().Set("Location", originURL)
 		res.WriteHeader(http.StatusTemporaryRedirect)
 	}
 }
+
+// GetURLs - возвращает оригинальные ссылки по передаваемым сокращенным ссылкам.
+func GetURLs(storage storage.Storage, baseURL string) http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+
+		// Аутентификация
+		userUID := fmt.Sprintf("%s", req.Context().Value(UserKeyUID))
+
+		var outputURLs []models.ResponseURL
+
+		allURLs, err := storage.FindByUserUID(userUID)
+		if err == nil {
+			for _, url := range allURLs {
+				outputURLs = append(outputURLs, models.ResponseURL{
+					OriginalURL: url.OriginalURL, ShortURL: fmt.Sprintf("%s/%s", baseURL, url.ShortHash),
+				})
+			}
+		}
```

Comment:
Стоит проверить ошибку и вернуть пользователю сообщение об ошибке, в противном случае получается, что ошибка игнорируется.

```suggestion
		if err != nil {
		    http.Error(res, "Error", http.StatusInternalServerError)
		}
	        for _, url := range allURLs {
		        outputURLs = append(outputURLs, models.ResponseURL{
			        OriginalURL: url.OriginalURL, ShortURL: fmt.Sprintf("%s/%s", baseURL, url.ShortHash),
		        })
	        }
```